# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/RemoteBrowser.h
# Commit: e65cb2d4c5a5
# Full Hash: e65cb2d4c5a55e3049922df02af643337b7a58b2
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 04:41:59
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 16 - Cycle collect RemoteBrowser and support weak references in BrowserHost. r=nika
#   
#   BrowserParent is cycle collected and supported weak references, so this commit adds support
#   for these things to BrowserHost.
#   
# ==============================================================================

diff -r 99f971a02d87 -r e65cb2d4c5a5 dom/ipc/RemoteBrowser.h
--- a/dom/ipc/RemoteBrowser.h	Wed May 08 16:12:26 2019 -0500
+++ b/dom/ipc/RemoteBrowser.h	Tue May 14 16:28:10 2019 -0500
@@ -11,6 +11,7 @@
 #include "mozilla/dom/ipc/IdType.h"
 #include "mozilla/layers/LayersTypes.h"
 #include "nsILoadContext.h"
+#include "nsISupports.h"
 #include "nsISupportsImpl.h"
 #include "nsIURI.h"
 #include "nsRect.h"
@@ -33,10 +34,8 @@
  * nsFrameLoader will use BrowserHost, and a content process nsFrameLoader will
  * use BrowserBridgeHost.
  */
-class RemoteBrowser {
+class RemoteBrowser : public nsISupports {
  public:
-  NS_INLINE_DECL_PURE_VIRTUAL_REFCOUNTING
-
   typedef mozilla::layers::LayersId LayersId;
 
   // Try to cast this RemoteBrowser to a BrowserHost, may return null
